docs/book/src/blockchain-development/external_code.md

```diff
@@ -0,0 +1,61 @@
+# External Code Execution
+
+The `std-lib` includes a function called `run_external` that allows Sway contracts to execute arbitrary external Sway code.
+
+This functionality enables features like upgradeable contracts and
+proxies.
+
+## Upgradeable Contracts
+
+Upgradeable contracts are designed to allow the logic of a smart contract to be updated after deployment.
+
+Consider this example target contract:
+
+```sway
+{{#include ../../../../examples/upgradeable_proxy/target-contract/src/main.sw:target}}
+```
+
+This contract has one function called `double_input`, which returns the input value times two.
+
+Below is what an upgradeable proxy contract could look like for this:
+
+```sway
+{{#include ../../../../examples/upgradeable_proxy/proxy-contract/src/main.sw:proxy}}
+```
+
+The contract has two functions:
+
+- `set_target_contract` updates the `target_contract` variable in storage with the `ContractId` of an external contract.
+- `double_input` reads the `target_contract` from storage and uses it to run external code. If the `target_contract` has a function with the same name (`double_input`), the code in the external `double_input` function will run.
+In this case, the function will return a `u64`.
+
+## Fallback functions
+
+If the function name doesn't exist in the target contract but a `fallback` function does, the `fallback` function will be triggered.
```

Comment:
What happens if there is zero or more than 1 fallback functions?

Reply:
Zero means the default behavior: revert on calling functions that aren't in the contract ABI.

More than one fallback function per contract produces an error.